packages/cli/src/commands/bundle/getAssetDestPathIOS.ts

```diff
@@ -12,7 +12,13 @@ import {PackagerAsset} from './assetPathUtils';
 function getAssetDestPathIOS(asset: PackagerAsset, scale: number): string {
   const suffix = scale === 1 ? '' : `@${scale}x`;
   const fileName = `${asset.name + suffix}.${asset.type}`;
-  return path.join(asset.httpServerLocation.substr(1), fileName);
+  return path.join(
+    // Assets can have relative paths outside of the project root.
+    // Replace `../` with `_` to make sure they don't end up outside of
```

Comment:
Is there a possibility where this could clash with existing assets that happen to be `_`-prefixed?

Reply:
There is, however if I remember correctly it would already cause issues on Android which has a similar replace function for paths.

Reply:
Yea, see the test I added for android without any code change to the logic.

https://github.com/react-native-community/cli/pull/939/files#diff-e43a39ec280faab0d111d719851dcc60R82

Reply:
Actually it is probably less likely to happen on Android since paths are flattened and all `/` are replaced with `_`. I think it is fine and if someone hits an issue we can figure out a fix. I don't think a lot of people are using this anyway since it's been broken forever.

Reply:
The part that concerns is that the fix would require coordination between RN and CLI releases. But I agree this should almost never happen (maybe we can warn somehow that "assets name clash, please rename"?)

Reply:
Yea I avoided changing the logic too much. I think ideally iOS should flatten the directory structure like android does. This would make it less likely (almost impossible) to happen. I think this would be a better long term solution but as you said would require coordination with RN.

Reply:
Since this is already kind of an edge case I'm more leaning towards wait and see if it's an issue before investing time in fixing it.